sample-demo-lambda-app/lambda_function.py

```diff
@@ -54,3 +54,39 @@ def check_prime(num):
             return False
         sq_root += 1
     return True
+
+
+def sync_ddb_table(source_ddb, destination_ddb):
+    response = source_ddb.scan(
+        TableName="table1"
+        )
+    for item in response['Items']:
+        destination_ddb.put_item(
+            TableName="table2", Item=item
+            )
+
+
+def list_sns(region, creds, sns_topics=[]):
+    sns = boto_session('sns', creds, region)
+    response = sns.list_topics()
+    for topic_arn in response["Topics"]:
+        sns_topics.append(topic_arn["TopicArn"])
+    return sns_topics
+  
+def process():
+    for region, creds in jobs["auth_config"]:
+        arns = list_sns(region, creds)
+        
+def read_lines(file):
+    lines = []
+    f = open(file, ‘r’)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *f*. The resource is allocated by call *builtins.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *f* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).